src/server/editorServices.ts

```diff
@@ -463,7 +448,45 @@ namespace ts.server {
                     this.typingsCache.deleteTypingsForProject(response.projectName);
                     break;
             }
-            project.updateGraph();
+            project.markAsDirty();
+            this.delayUpdateProjectGraphAndInferredProjectsRefresh(project);
+        }
+
+        private delayInferredProjectsRefresh() {
+            this.pendingInferredProjectUpdate = true;
+            this.throttledOperations.schedule("*refreshInferredProjects*", /*delay*/ 250, () => {
+                if (this.pendingProjectUpdates.size !== 0) {
+                    this.delayInferredProjectsRefresh();
+                }
+                else if (this.pendingInferredProjectUpdate) {
+                    this.pendingInferredProjectUpdate = false;
+                    this.refreshInferredProjects();
+                }
+            });
+        }
+
+        private delayUpdateProjectGraph(project: Project) {
+            const projectName = project.getProjectName();
+            this.pendingProjectUpdates.set(projectName, project);
+            this.throttledOperations.schedule(projectName, /*delay*/ 250, () => {
+                const project = this.pendingProjectUpdates.get(projectName);
+                if (project) {
+                    this.pendingProjectUpdates.delete(projectName);
+                    project.updateGraph();
+                }
+            });
+        }
+
+        delayUpdateProjectGraphAndInferredProjectsRefresh(project: Project) {
+            this.delayUpdateProjectGraph(project);
+            this.delayInferredProjectsRefresh();
+        }
+
+        private delayUpdateProjectGraphs(projects: Project[]) {

```

Comment:
Not really. RefreshInferredProjects goes through all open files and assign them to inferred project if they are not part of any project. Updating graph of the inferred projects after that will ensure that inferred projects are upto date. 

We want to update graphs of existing inferred projects (when projects === this.inferrredProjects  when called (eg. to ensure that inferred project graphs take effect of updated compiler options and files are removed from these projects/ added to them) after that we want to refresh inferred project to actually make sure all the open files have correct project assignment